tests/drum/test_units.py

```diff
@@ -185,6 +204,51 @@ def training_metadata_yaml_with_proj():
 }
 
 
+@pytest.mark.parametrize(
+    "config_yaml",
+    [
+        "custom_predictor_metadata_yaml",
+        "training_metadata_yaml",
+        "training_metadata_yaml_with_proj",
+        "inference_metadata_yaml",
+        "inference_multiclass_metadata_yaml",
+        "inference_multiclass_metadata_yaml_label_file",
+    ],
+)
+@pytest.mark.parametrize("existing_model_id", [None, modelID])
+def test_yaml_metadata(request, config_yaml, existing_model_id):
+    config_yaml = request.getfixturevalue(config_yaml)
+    if existing_model_id:
+        config_yaml = config_yaml + "\nmodelID: {}".format(existing_model_id)
+
+    try:
+        temp_dir = tempfile.mkdtemp()
+        with open(os.path.join(temp_dir, MODEL_CONFIG_FILENAME), mode="w") as f:
+            f.write(config_yaml)
+        read_model_metadata_yaml(temp_dir)
+    finally:
+        shutil.rmtree(temp_dir)
+
+
+def test_yaml_metadata_missing_fields(custom_predictor_metadata_yaml):
+    try:
+        temp_dir = tempfile.mkdtemp()
```

Comment:
use pytest's `tmp_path` fixture -- pytest will take care of the cleanup
https://docs.pytest.org/en/stable/tmpdir.html